lib/launcher.rb

```diff
@@ -119,18 +119,18 @@ def webrick_server(app)
     end
 
     def launch
-      ::Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
-
-      http_app = http_app()
-      https_app = https_app()
-      raise Exception.new("Both http and https are disabled, unable to start.") if http_app.nil? && https_app.nil?
-
       if SETTINGS.daemon
         check_pid
         Process.daemon
         write_pid
       end
 
+      ::Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
+
+      http_app = http_app()
+      https_app = https_app()
+      raise Exception.new("Both http and https are disabled, unable to start.") if http_app.nil? && https_app.nil?
```

Comment:
If the check's done before forking, executing bin/smart-proxy shows this exception on stdout and exits with a non-zero exit code which service managers like systemd and even init scripts will pick up immediately. If it moves to after forking, then it forks away, but the process stops and the error's only shown in the log.